services/git-service-api/src/main/java/io/fabric8/launcher/service/git/OAuthTokenProvider.java

```diff
@@ -0,0 +1,23 @@
+package io.fabric8.launcher.service.git;
+
+import io.fabric8.launcher.service.git.api.GitServiceConfig;
+
+/**
+ * Fetch oauth tokens for git providers
+ */
+public interface OAuthTokenProvider {
```

Comment:
Maybe we can make it an utility class instead? Or will we have different implementations for this interface?

Reply:
How would you inject things if it just a static class? It seems all services have one implementation, it would require us to get rid of the `git-service-impl` module?

Reply:
Sorry, not utility class with static methods, but a single class with `@ApplicationScoped` and constructor injection.

What do you mean by all services having one implementation? AFAIK `GitService` has multiple implementations

Reply:
What I mean is all modules are split into 2 one is called `<module>-api` the other `<module>-impl`. Consuming modules only deal with what is in `<module>-api` and they "know" nothing about the `<module>-impl` right? So if we don't want an interface we need to move this implementing class to the `git-service-api` module (or should it be somewhere else)? An example of what I mean is `BoosterCatalogFactory` it's in `core-api` and only has one implementation `RhoarBoosterCatalogFactory` in `core-impl`. So if this reasoning is completely wrong then maybe we need to add it to the contribution guide, with some clear guides to when to do the split and when to create an "utility class"